### PR TITLE
[Backend] Implemented input validation for post requests

### LINKED ISSUE #71: [Backend] input validation missing in POST routes
**Describe the bug**
The post routes in the backend don't carry out input validation, due to which if any values are missing, it just gives a server error response.

**To Reproduce**
Execute any post request without providing the required data

**Expected behavior**
It should give a clear error message

**Screenshots**

screen-capture

**Desktop (please complete the following information):**
 - Windows 10

### PR DESCRIPTION
**Pull request by amlannandy**
**Description**

Implemented input validation in post routes by adding the following measures -
1. If not post data is sent at all
2. If some data is sent but not the fields that are required

**Demo**

screen-capture (1)

fixes #71

### CODE REVIEW

**Comment by shivanshuraj1333** [comment]
@amlannandy please resolve conflicts

**Comment by amlannandy** [comment]
@shivanshu1333 Done. As we're now only deleting the current user, email is not necessary. So I removed the input validation.

File: flask-backend/api/routes/user.py
Comment (shivanshuraj1333) [inline flask-backend/api/routes/user.py]: this API will be used by the admin to delete any particular user, we're not supposed to delete the current user here.

Comment (amlannandy) [reply flask-backend/api/routes/user.py]: @shivanshu1333 Yeah that's what I was also thinking. You approved #68  so I thought we're changing the functionality. I'll change it back.

Comment (sachinsom93) [reply flask-backend/api/routes/user.py]: Hi @shivanshu1333, @amlannandy, 
Previously I also thought that only admin can delete any of his user, but at that time some funtionality related to admin was not created like create user by admin, route to show all users of admin, so thought to change delete function in such a way that only a user can delete only his account as previously any user can delete anybody's account.
I have also mentioned a issue related to this, I'll soon raise a PR for that, it would contain route for only admin can create users and delete his users.

Comment (amlannandy) [reply flask-backend/api/routes/user.py]: @sachinsom93 I see. Sounds good. 👍🏻

**Comment by amlannandy** [comment]
@shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.

**Comment by sachinsom93** [comment]
> @shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.

Hi @amlannandy, I have raised PR #76 for admin related funtionality where I've created a "remove-user" route for  deleting a user by his admin, so I think we don't need this delete route. We have to wait for @shivanshu1333 's approval for PR #76.

**Comment by amlannandy** [comment]
> > @shivanshu1333 I have changed the delete route. Do let me know if it works or if any other change is required.
> 
> Hi @amlannandy, I have raised PR #76 for admin related funtionality where I've created a "remove-user" route for deleting a user by his admin, so I think we don't need this delete route. We have to wait for @shivanshu1333 's approval for PR #76.

Gotcha. I have just reverted the change as I was asked. Will undo the last commit if needed. :)

**Comment by shivanshuraj1333** [comment]
Please resolve conflicts

**Comment by amlannandy** [comment]
> Please resolve conflicts

Done 👍🏻

**Review by shivanshuraj1333 — APPROVED** [review]

**Pull request merged**
